runtimeconfig/manager_test.go

```diff
@@ -76,27 +75,171 @@ func newTestOverridesManagerConfig(t *testing.T, i int32) (*atomic.Int32, Config
 	}
 }
 
+func generateRuntimeFiles(overrideStrings []string) ([]*os.File, error) {
+	var overrideFiles []*os.File
+
+	for count, override := range overrideStrings {
+		pattern := fmt.Sprintf("overrides-file-%d", count)
+		tempFile, err := os.CreateTemp("", pattern)
+		if err != nil {
+			return nil, err
+		}
+		_, err = tempFile.WriteString(override)
+		if err != nil {
+			return nil, err
+		}
+		overrideFiles = append(overrideFiles, tempFile)
+	}
+	return overrideFiles, nil
+}
+
+func generateLoadPath(overrideFiles []*os.File) string {
+	var fileNames []string
+	for _, f := range overrideFiles {
+		fileNames = append(fileNames, f.Name())
+	}
+	return strings.Join(fileNames, ",")
+}
+
+func cleanup(overrideFiles []*os.File) error {
+	for _, f := range overrideFiles {
+		err := f.Close()
+		if err != nil {
+			return err
+		}
+		err = os.Remove(f.Name())
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 func TestNewOverridesManager(t *testing.T) {
-	tempFile, err := os.CreateTemp("", "test-validation")
+	tempFiles, err := generateRuntimeFiles(
+		[]string{`overrides:
+  user1:
+    limit2: 150`})
 	require.NoError(t, err)
 
 	defer func() {
-		// Clean up
-		require.NoError(t, tempFile.Close())
-		require.NoError(t, os.Remove(tempFile.Name()))
+		require.NoError(t, cleanup(tempFiles))
 	}()
 
-	_, err = tempFile.WriteString(`overrides:
+	defaultTestLimits = &TestLimits{Limit1: 100}
+
+	// testing NewRuntimeConfigManager with overrides reload config set
+	overridesManagerConfig := Config{
+		ReloadPeriod: time.Second,
+		LoadPath:     generateLoadPath(tempFiles),
+		Loader:       testLoadOverrides,
+	}
+
+	overridesManager, err := New(overridesManagerConfig, nil, log.NewNopLogger())
+	require.NoError(t, err)
+	require.NoError(t, services.StartAndAwaitRunning(context.Background(), overridesManager))
+
+	// Cleaning up
+	require.NoError(t, services.StopAndAwaitTerminated(context.Background(), overridesManager))
+
+	// Make sure test limits were loaded.
+	require.NotNil(t, overridesManager.GetConfig())
```

Comment:
Would it be worth it to also check the values?
```suggestion
	require.NotNil(t, overridesManager.GetConfig())
	conf := overridesManager.GetConfig().(*testOverrides)
	require.Equal(t, 150, conf.Overrides["user1"].Limit2)
	require.NotNil(t, overridesManager.GetConfig())
```